include/rtc/rtcp.hpp

```diff
@@ -20,35 +20,52 @@
 #ifndef RTC_RTCP_H
 #define RTC_RTCP_H
 
+#include <utility>
+
 #include "include.hpp"
 #include "log.hpp"
 #include "message.hpp"
+#include "rtp.hpp"
 
 namespace rtc {
 
-typedef uint32_t SSRC;
-
 class RtcpHandler {
 public:
-	virtual void onOutgoing(std::function<void(rtc::message_ptr)> cb) = 0;
-	virtual std::optional<rtc::message_ptr> incoming(rtc::message_ptr ptr) = 0;
+    /**
+     * If there is traffic coming from the remote side
+     * @param ptr
+     * @return
+     */
+    virtual rtc::message_ptr incoming(rtc::message_ptr ptr) = 0;
+
+    /**
+     * If there is traffic being sent to the remote side
+     * @param ptr
+     * @return
+     */
+    virtual rtc::message_ptr outgoing(rtc::message_ptr ptr) = 0;
 };
 
+class Track;
+
 // An RtcpSession can be plugged into a Track to handle the whole RTCP session
-class RtcpSession : public RtcpHandler {
+class RtcpReceivingSession : public RtcpHandler {
+protected:
+    std::shared_ptr<Track> track;
 public:
-	void onOutgoing(std::function<void(rtc::message_ptr)> cb) override;
+    RtcpReceivingSession(std::shared_ptr<Track> track): track(std::move(track)) {}
```

Comment:
I have actually added the `RtcpSession::onOutgoing` callback to prevent that circular dependency. Here, `Track` has a shared pointer on `RtcpHandler` and `RtcpHandler` has a backward shared pointer on `Track`, so neither will be destructed.